spec/controllers/single_page_subscriptions_controller_spec.rb

```diff
@@ -0,0 +1,115 @@
+RSpec.describe SinglePageSubscriptionsController do
+  include GdsApi::TestHelpers::ContentStore
+  include GdsApi::TestHelpers::AccountApi
+  include GdsApi::TestHelpers::EmailAlertApi
+  include GovukPersonalisation::TestHelpers::Requests
+
+  render_views
+
+  let(:topic) { "/test" }
+  let(:topic_slug) { SecureRandom.uuid }
+  let(:topic_name) { "Test" }
+  let(:redirect_path) { "/email/subscriptions/account/confirm?topic=#{topic}" }
+  let(:auth_provider) { "http://auth/provider" }
+  let(:params) { { topic: topic } }
+
+  describe "when feature flag is not 'enabled'" do
+    it "POST /email/subscriptions/single-page/new-session returns 404" do
+      post :edit
+      expect(response).to have_http_status(:not_found)
+    end
+
+    it "POST /email/subscriptions/single-page/new" do
+      post :show
+      expect(response).to have_http_status(:not_found)
+    end
+  end
+
+  context "when the feature is on" do
+    around do |example|
+      ClimateControl.modify FEATURE_FLAG_GOVUK_ACCOUNT: "enabled" do
+        example.run
+      end
+    end
+
+    describe "POST /email/subscriptions/single-page/new-session" do
+      before { stub_account_api_get_sign_in_url(auth_uri: auth_provider, redirect_path: redirect_path) }
+
+      it "redirects to sign in with a redirect_path param" do
+        post :edit, params: params.merge({ redirect_path: redirect_path })
```

Comment:
Why are you passing the `redirect_path` as a parameter here?

Reply:
Good spot, I think this is left over from some earlier testing